lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1

```diff
@@ -209,8 +209,10 @@ Function Get-AnsibleParam($obj, $name, $default = $null, $resultobj = @{}, $fail
         if ($type -eq "path") {
             # Expand environment variables on path-type
             $value = Expand-Environment($value)
-            # Test if a valid path is provided
-            if (-not (Test-Path -IsValid $value)) {
+            # Test-Path -IsValid failed when the long path prefix is there, we don't want
+            # to check this ourselves and just rely on Windows to validate the path when
+            # using it
+            if ((-not ($value.StartsWith("\\?\"))) -and (-not (Test-Path -Path $value -IsValid))) {
```

Comment:
I'm wondering whether we should not even run `Expand-Environment` on the value if it starts with `\\?\` to allow people to force a file/dir name with a value that's like an env var.